nvflare/private/fed/simulator/simulator_server.py

```diff
@@ -0,0 +1,115 @@
+# Copyright (c) 2021-2022, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import List, Optional
+
+from nvflare.apis.fl_component import FLComponent
+from nvflare.apis.fl_constant import FLContextKey, RunProcessKey, ServerCommandKey
+from nvflare.apis.fl_context import FLContext
+from nvflare.apis.shareable import ReturnCode, Shareable, make_reply
+from nvflare.private.fed.server.server_state import HotState
+
+from ..server.fed_server import FederatedServer
+from ..server.server_engine import ServerEngine
+
+
+class SimulatorServerEngine(ServerEngine):
+    def persist_components(self, fl_ctx: FLContext, completed: bool):
+        pass
+
+    def sync_clients_from_main_process(self):
+        pass
+
+    def parent_aux_send(self, targets: [], topic: str, request: Shareable, timeout: float, fl_ctx: FLContext) -> dict:
+        replies = self.aux_send(targets=targets, topic=topic, request=request, timeout=timeout, fl_ctx=fl_ctx)
+
+        return replies
+
+
+class SimulatorServer(FederatedServer):
```

Comment:
should this be `SimulatorFederatedServer`, although the name makes not much sense, but from the name we can tell this is inherited from FederatedServer.

Otherwise SimulatorServer is ambiguous?